doc/stdlib/base.rst

```diff
@@ -983,61 +1460,120 @@ System
 
 .. function:: gethostname() -> AbstractString
 
+   .. Docstring generated from Julia source
+   .. code-block:: julia
+       gethostname() -> AbstractString
+
    Get the local machine's host name.
 
 .. function:: getipaddr() -> AbstractString
 
+   .. Docstring generated from Julia source
+   .. code-block:: julia
+       getipaddr() -> AbstractString
+
    Get the IP address of the local machine, as a string of the form "x.x.x.x".
 
 .. function:: getpid() -> Int32
 
+   .. Docstring generated from Julia source
+   .. code-block:: julia
+       getpid() -> Int32
+
    Get julia's process ID.
 
 .. function:: time()
 
+   .. Docstring generated from Julia source
+   .. code-block:: julia
+       time()
+
    Get the system time in seconds since the epoch, with fairly high (typically, microsecond) resolution.
 
 .. function:: time_ns()
 
+   .. Docstring generated from Julia source
+   .. code-block:: julia
+       time_ns()
+
    Get the time in nanoseconds. The time corresponding to 0 is undefined, and wraps every 5.8 years.
 
 .. function:: tic()
 
+   .. Docstring generated from Julia source
+   ::
+
+              tic()
+
    Set a timer to be read by the next call to :func:`toc` or :func:`toq`. The macro call ``@time expr`` can also be used to time evaluation.
 
 .. function:: toc()
 
+   .. Docstring generated from Julia source
+   ::
+
+              toc()
+
    Print and return the time elapsed since the last :func:`tic`.
 
 .. function:: toq()
 
+   .. Docstring generated from Julia source
+   ::
+
+              toq()
+
    Return, but do not print, the time elapsed since the last :func:`tic`.
 
 .. function:: @time
 
-   A macro to execute an expression, printing the time it took to execute, the number of allocations, and the total number of bytes its execution caused to be allocated, before returning the value of the expression.
+   .. Docstring generated from Julia source
+   Executes an expression, printing the time it took to execute and the total number of bytes its execution caused to be allocated. Returns the value of the expression. For example:
```

Comment:
Also being overwritten by basedocs. Basedocs is out of date in not mentioning the number of allocations, but does have a nice example. Would be good to combine the two.